tobac/feature_detection.py

```diff
@@ -1121,39 +1121,30 @@ def feature_detection_multithreshold_timestep(
         elif i_threshold == 0:
             regions_old = regions_i
 
-        if statistic:
-            # reconstruct the labeled regions based on the regions dict
-            labels = np.zeros(track_data.shape)
-            labels = labels.astype(int)
-            for key in regions_old.keys():
-                labels.ravel()[regions_old[key]] = key
-                # apply function to get statistics based on labeled regions and functions provided by the user
-                # the feature dataframe is updated by appending a column for each metric
-            if statistics_unsmoothed:
```

Comment:
I just noted that the optional calculation on the unsmoothed input data is not merged into your PR yet and since you removed a whole paragraph and placed it somewhere else I bet that these lines would disappear. Maybe the easiest fix is to copy over `if statistics_unsmoothed` and place it within the code paragraph you added after each threshold has been calculated?

Reply:
Good catch! I'll add that back in